In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("logs.csv")

In [4]:
selected_df = df[['lambda1', 'lambda2', 'best_val_rmse','best_val_accuracy','test.accuracy','test.rmse','train/loss_ce','train/loss_mse']]


In [5]:
selected_df.head()

lambda1  lambda2  best_val_rmse  best_val_accuracy  test.accuracy  \
0      0.5      1.0       0.071452           0.890833         0.8815   
1      1.0      0.1       0.115397           0.900500         0.8891   
2      1.0      2.0       0.078293           0.885000         0.8871   
3      2.0      1.0       0.100364           0.895667         0.8956   
4      1.0      1.0       0.128402           0.895000         0.8904   

   test.rmse  train/loss_ce  train/loss_mse  
0   0.167839       0.375371        0.051169  
1   0.153857       0.336763        0.060027  
2   0.090649       0.377883        0.064387  
3   0.160150       0.349619        0.057395  
4   0.143256       0.375131        0.098874

---

## 📊 Quick summary of your results

| λ₁  | λ₂  | Test Acc     | Test RMSE    | Insight                                          |
| --- | --- | ------------ | ------------ | ------------------------------------------------ |
| 0.5 | 1.0 | **0.8815**   | 0.1678       | Regression dominates → worse accuracy            |
| 1.0 | 0.1 | 0.8891       | 0.1539       | Classification priority → good acc + decent rmse |
| 1.0 | 2.0 | 0.8871       | **0.0906** ✅ | Regression strong → best RMSE                    |
| 2.0 | 1.0 | **0.8956** ✅ | 0.1601       | Best overall accuracy                            |
| 1.0 | 1.0 | 0.8904       | 0.1433       | Balanced baseline, not the best in either        |

---

##  Key Insights

###  1️⃣ Classification improves when λ₁ > λ₂

* Best accuracy at **λ₁ = 2.0, λ₂ = 1.0**
* Also strong at **λ₁ = 1.0, λ₂ = 0.1**

 When classification is weighted more → model learns better class boundaries
 But regression metrics degrade → RMSE around 0.15–0.16

---

###  2️⃣ Regression improves when λ₂ > λ₁

* Best RMSE: **λ₁ = 1.0, λ₂ = 2.0 → 0.0906**

 When regression has more weight → numerical prediction gets precise
📉 Accuracy slightly drops

---

###  3️⃣ Equal weighting is **suboptimal**

* λ₁ = 1.0, λ₂ = 1.0 → results are mediocre in both tasks
  → Confirms the tasks interfere → requires trade-off tuning

---

## 🔄 Loss values confirm gradients dominance:

| Run           | λ₁ vs λ₂ relationship      | train CE  | train MSE | Meaning                  |
| ------------- | -------------------------- | --------- | --------- | ------------------------ |
| High λ₂ (1,2) | CE high → not prioritized  | **0.378** | 0.064     | Better regression      |
| High λ₁ (2,1) | MSE high → not prioritized | **0.350** | 0.057     | Better classification  |

So the model *really* follows whichever task you emphasize.

---

##  Conclusion

| Goal                      | Best λ settings                   |
| ------------------------- | --------------------------------- |
| Prioritize Classification | **λ₁ = 2.0, λ₂ = 1.0**            |
| Prioritize Regression     | **λ₁ = 1.0, λ₂ = 2.0**            |
| Balanced?                 | Neither — need adaptive weighting |

You have a **Pareto trade-off**:
No single λ gives the best score in both tasks simultaneously.

---



### The model with the highest validation classification accuracy

In [6]:
df.sort_values(by="best_val_accuracy", ascending=False).head(1)


Name Notes  Tags                   Created  Runtime  Sweep  \
6  haunted-ritual-16     -   NaN  2025-10-30T11:24:39.000Z      578    NaN   

   batch_size  dropout_rate  epochs  lambda1  ...  test.rmse  train/loss_ce  \
6         256           0.1      10      1.0  ...   0.130828       0.309851   

  train/loss_mse train/loss_total  val/accuracy  val/loss_ce  val/loss_mse  \
6       0.026386         0.312489      0.899667     0.269275      0.013842   

   val/loss_total   val/mae  val/rmse  
6        0.270659  0.091332  0.117651  

[1 rows x 32 columns]

### The model with the lowest validation regression RMSE

In [7]:
df.sort_values(by="best_val_rmse", ascending=True).head(1)


Name Notes  Tags                   Created  Runtime  Sweep  \
0  dauntless-cloud-22     -   NaN  2025-11-03T00:15:31.000Z      660    NaN   

   batch_size  dropout_rate  epochs  lambda1  ...  test.rmse  train/loss_ce  \
0         128          0.25      10      0.5  ...   0.167839       0.375371   

  train/loss_mse train/loss_total  val/accuracy  val/loss_ce  val/loss_mse  \
0       0.051169         0.238855      0.876167     0.329645      0.022311   

   val/loss_total   val/mae  val/rmse  
0        0.187133  0.124286  0.149369  

[1 rows x 32 columns]


### 